### ISSUE TITLE
Java shell exec does not return after selenium calls on accessing /dev/urandom

### ISSUE THREAD

**Issue by crocodilechris** [SeleniumHQ/selenium#732]
Hi,

We use a couple of different "connectors" to assemble our tests, so the example code is a bit artificial, but the problem is there...
I make a java shell exec on bash and it works fine, except when combining it with selenium -- then the waitFor() method does not return. The problem occurs if I try to read from a system device. If I call the command chain as unprivileged user, I get a result, so it can't be an problem with the permissions.
It makes no difference if I use a WebDriver or a RemoteWebDriver.

The class was run in a fresh eclipse project with the pom.xml provided in the documentation of selenium (2.46.0). 
Java version:
 ```
java version "1.7.0_79"
OpenJDK Runtime Environment (IcedTea 2.5.5) (7u79-2.5.5-0ubuntu0.14.04.2)
OpenJDK 64-Bit Server VM (build 24.79-b02, mixed mode)
```
Minimal example:
```java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Testing {
	public static void main(String[] args) {
		WebDriver driver = null;
		driver = new FirefoxDriver();
		System.out.println("Start");
		driver.get("http://www.seleniumhq.org/");
		System.out.println("After get");
		driver.quit();
		System.out.println("Browser closed");

		// works
		// String[] cmd = { "/bin/bash", "-c", "echo 'DS%&/%s56Df%&43dsg' | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1" }; // works
		// works not
		String[] cmd = { "/bin/bash", "-c", "cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1" }; // works not
		StringBuffer buf = new StringBuffer();

		System.out.println("Call: " + Arrays.toString(cmd));

		try {
			Process proc = Runtime.getRuntime().exec(cmd);

			System.out.println("Now waitFor process ...");
			proc.waitFor();
			System.out.println("Process returned.");

			BufferedReader reader = new BufferedReader(new InputStreamReader(proc.getInputStream()));

			String line = "";
			while ((line = reader.readLine()) != null)
			{
				buf.append(line + "\n");
			}
			System.out.println("Buf: " + buf.toString());
		} catch (Exception e) {
			System.out.println("ERROR");
		}
	}
}
```

**Comment by nmondal** [comment]
Hi, This is a problem in waitFor() or rather more importantly 
![this link](http://stackoverflow.com/questions/5483830/process-waitfor-never-returns)
I do not think - this is selenium specific.

**Comment by ddavison** [comment]
as soon as you reach your `String[] cmd...` bit.. it's out of Selenium's jurisdiction.  This is not a problem with Selenium, this is a help request.

GitHub issues are not a forum for Q&A help requests - please use the [selenium users group](https://groups.google.com/forum/#!forum/selenium-users)

Closing.

**Comment by crocodileandy** [comment]
If you remove the code containing the Selenium stuff, and start with the String cmd[] line the code works fine. So it seems as it has to do something with Selenium. Otherwise the code had to show the same blocking effect.

**Comment by crocodileandy** [comment]
Could you please reopen this issue, because I think its problem of Selenium. 
See my last comment about this.

**Comment by lukeis** [comment]
@crocodileandy I went ahead and attempted this and was able to reproduce your test case and saw that removing the selenium pieces then caused it to succeed when running against OpenJDK.

So then I switched my Ubuntu instance from OpenJDK to Oracle's JDK... and then it magically worked, including the bits with running on selenium.

This doesn't seem to be a selenium bug, rather a bug in OpenJDK. I do not understand the complexities happening here (nor do I have the time to attempt to do a deep dive into this). Hope this helps, but there's really nothing Selenium can do here.
